app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java

```diff
@@ -542,6 +546,43 @@ public static Intent getPlayQueueActivityIntent(final Context context) {
         return intent;
     }
 
+    /*//////////////////////////////////////////////////////////////////////////
+    // Downloading a video without starting a player
+    //////////////////////////////////////////////////////////////////////////*/
+
+    public static void openDownloadDialogWithoutStartingPlayer(
+            final AppCompatActivity activity, final StreamInfo currentInfo) {
+        if (currentInfo == null) {
+            return;
+        }
+
+        try {
+            // Get the sortedVideoStreams and selectedVideoStreamIndex using ListHelper
+            final ArrayList<VideoStream> sortedVideoStreams = new ArrayList<>(
```

Comment:
The variable's type should be `List<VideoStream>`, as we do not care about the implementation:
```suggestion
            final List<VideoStream> sortedVideoStreams = new ArrayList<>(
```

Reply:
Made the changes that you suggested. However, using the same code in `VideoDetailFragment` will break `MainActivity`
https://github.com/TeamNewPipe/NewPipe/blob/484c852efd55eb2d0279c01d19529ea5b9d59f0a/app/src/main/java/org/schabi/newpipe/MainActivity.java#L619-L621